ding/model/template/vac.py

```diff
@@ -63,40 +63,56 @@ def __init__(
             else:
-                raise RuntimeError(
-                    "not support obs_shape for pre-defined encoder: {}, please customize your own DQN".
-                    format(obs_shape)
-                )
-            self.share_encoder = share_encoder
-            if self.share_encoder:
-                if self.impala_cnn_encoder:
-                    self.encoder = ImpalaConvEncoder(obs_shape)
-                else:
-                    self.encoder = encoder_cls(
-                        obs_shape, encoder_hidden_size_list, activation=activation, norm_type=norm_type
+                if isinstance(obs_shape, int) or len(obs_shape) == 1:
+                    return FCEncoder(
+                        obs_shape=obs_shape,
+                        hidden_size_list=encoder_hidden_size_list,
+                        activation=activation,
+                        norm_type=norm_type
                     )
-            else:
-                if self.impala_cnn_encoder:
-                    self.actor_encoder = ImpalaConvEncoder(obs_shape)
-                    self.critic_encoder = ImpalaConvEncoder(obs_shape)
-                else:
-                    self.actor_encoder = encoder_cls(
-                        obs_shape, encoder_hidden_size_list, activation=activation, norm_type=norm_type
+                elif len(obs_shape) == 3:
+                    return ConvEncoder(
+                        obs_shape=obs_shape,
+                        hidden_size_list=encoder_hidden_size_list,
+                        activation=activation,
+                        norm_type=norm_type
                     )
-                    self.critic_encoder = encoder_cls(
-                        obs_shape, encoder_hidden_size_list, activation=activation, norm_type=norm_type
+                else:
+                    raise RuntimeError(
+                        "not support obs_shape for pre-defined encoder: {}, please customize your own encoder".
+                        format(obs_shape)
                     )
+
+        if self.share_encoder:
+            assert actor_head_hidden_size == critic_head_hidden_size, \
+                "actor and critic network head should have same size."
+            if encoder:
+                if type(encoder) is torch.nn.Module:
+                    self.encoder = encoder
+                else:
+                    raise ValueError("illegal encoder instance.")
+            else:
+                self.encoder = new_encoder(actor_head_hidden_size)
         else:
-            assert share_encoder
-            self.share_encoder = share_encoder
-            self.encoder = encoder
+            if encoder:
+                if type(encoder) is torch.nn.Module:
+                    raise ValueError(
+                        "one encoder instance is not allowed \
+                        to be assigned to actor critic that not sharing encoders."
+                    )
+                else:
```

Comment:
where is you encoder assignment branch, and add corresponding unittest

Reply:
fixed.
unittest for encoder assignment is added.